src/prover_disk.hpp

```diff
@@ -326,8 +326,10 @@ class DiskProver {
         std::vector<uint8_t> deltas =
             Encoding::ANSDecodeDeltas(bit_mask, encoded_size, kCheckpoint1Interval, kC3R);
         std::vector<uint64_t> p7_positions;
+        bool surpassed_f7 = false;
         for (uint8_t delta : deltas) {
             if (curr_f7 > f7) {
+                surpassed_f7 = true;
```

Comment:
perhaps I'm missing something, but it looks like we could just return `p2_positions` here, remove the `surpassed_f7` boolean and return an empty vector unconditionally at the end of the function.

Reply:
Yeah that is simpler..